comp/trace/compression/impl-zstd/zstd.go

```diff
@@ -0,0 +1,34 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build zstd
+
```

Comment:
No need for build tag

Reply:
why not? this module has a dependency on zstd, unless explicitly controlled with build tags, you will still face compile errors on systems without cgo

Reply:
I have updated this PR removing the build tag, but this has required a refactor on how compressor is used, and now it can be passed to the agent/writer. This is the only way to avoid having build tags